Navigation Component: Update docs (#24880)
WordPress/gutenberg: packages/components/src/navigation/README.md

# Navigation

> Render a heirarchy of menu items into a waterfall style navigation.

Reviewer: Great description.  One small typo ("heirarchy") and optional, but should we include "flat" to help signal the data structure used?

```suggestion
Render a flat array of menu items into a waterfall style hierarchy navigation.
```

I'm not sure if that makes things more clear or not; will leave that up to your discretion.

## Usage

```jsx

> import Navigation from '@wordpress/components';

Reviewer: I don't think we can import from the default and the names may include "experimental" for a while; not sure if worth noting here:

```suggestion
import {
	__experimentalNavigation as Navigation,
	__experimentalNavigationMenu as NavigationMenu,
	__experimentalNavigationMenuItem as NavigationMenuItem,
} from '@wordpress/components';
```

Author: Probably a good idea. There are instances of each present in the codebase.

https://github.com/WordPress/gutenberg/blob/ebd5d9605f0aa57088f74d0ecaccf93017e36d5e/packages/components/src/number-control/README.md#L8

https://github.com/WordPress/gutenberg/blob/ebd5d9605f0aa57088f74d0ecaccf93017e36d5e/packages/components/src/text/README.md#L8

### `data`

-   Type: `array`
-   Required: Yes

An array of config objects for each menu item.

> Config objects can be represented

Reviewer: Missing `onClick` param option.

#### `config.title`

-   Type: `string`
-   Required: Yes

A menu item's title.

#### `config.id`

-   Type: `string|Number`
-   Required: Yes

A menu item's id.

#### `config.parent`

-   Type: `string|Number`
-   Required: No

> Specify a menu item's parent id.

Reviewer: Maybe note that `root` will get added as the menu item's parent if none is provided.

#### `config.href`

-   Type: `string`
-   Required: No

Turn a menu item into a link by supplying a url.

#### `config.linkProps`

-   Type: `object`
-   Required: No

> Supply an properties passed to the menu-item.

Reviewer: ```suggestion
Supply properties passed to the menu-item.
```
